net-mail/cyrus-imapd/cyrus-imapd-3.4.9.ebuild

```diff
@@ -0,0 +1,264 @@
+		$(use_with clamav) \
+		$(use_enable nntp) \
+		$(use_enable http) \
+		$(use_with http nghttp2) \
+		$(use_enable replication) \
+		$(use_enable kerberos gssapi) \
+		$(use_with ldap) \
+		$(use_with mysql) \
+		$(use_with postgres pgsql) \
+		$(use_with perl) \
+		$(use_with sqlite) \
+		$(use_with ssl openssl) \
+		$(use_enable server) \
+		$(use_enable sieve) \
+		$(use_enable static-libs static) \
+		$(use_with tcpd libwrap) \
+		$(use_enable xapian) \
+		$(use_enable test unit-tests) \
+		${myconf}
+}
+
+src_install() {
+	emake DESTDIR="${D}" INSTALLDIRS=vendor install
+
+	dodoc README*
+	dodoc -r doc
+
+	cp -r contrib tools "${ED}"/usr/share/doc/${PF} || die
+	rm -f doc/text/Makefile* || die
+
+	mv "${ED}"/usr/libexec/{master,cyrusmaster} || die
+
+	insinto /etc
+	newins "${ED}"/usr/share/doc/${PF}/doc/examples/cyrus_conf/normal.conf cyrus.conf
+	newins "${ED}"/usr/share/doc/${PF}/doc/examples/imapd_conf/normal.conf imapd.conf
+
+	sed -i -e '/^configdirectory/s|/var/.*|/var/imap|' \
+		-e '/^partition-default/s|/var/.*|/var/spool/imap|' \
+		-e '/^sievedir/s|/var/.*|/var/imap/sieve|' \
+		"${ED}"/etc/imapd.conf || die
+
+	sed -i -e 's|/var/imap/socket/lmtp|/run/cyrus/socket/lmtp|' \
+		-e 's|/var/imap/socket/notify|/run/cyrus/socket/notify|' \
+		"${ED}"/etc/cyrus.conf || die
+
+	# turn off sieve if not installed
+	if ! use sieve; then
+		sed -i -e "/sieve/s/^/#/" "${ED}"/etc/cyrus.conf || die
+	fi
+
+	# same thing for http(s) as well
+	if ! use http; then
+		sed -i -e "/http/s/^/#/" "${ED}"/etc/cyrus.conf || die
+	fi
+
+	newinitd "${FILESDIR}"/cyrus.rc8 cyrus
+	newconfd "${FILESDIR}"/cyrus.confd cyrus
+	newpamd "${FILESDIR}"/cyrus.pam-include sieve
+
+	for subdir in imap/{,db,log,msg,proc,socket,sieve} spool/imap/{,stage.} ; do
+		keepdir "/var/${subdir}"
+		fowners cyrus:mail "/var/${subdir}"
+		fperms 0750 "/var/${subdir}"
+	done
+
+	for subdir in imap/{user,quota,sieve} spool/imap ; do
+		for i in a b c d e f g h i j k l m n o p q r s t v u w x y z ; do
+			keepdir "/var/${subdir}/${i}"
+			fowners cyrus:mail "/var/${subdir}/${i}"
+			fperms 0750 "/var/${subdir}/${i}"
+		done
+	done
+
+	find "${ED}" -type f -name '*.la' -delete || die
+}
+
+pkg_preinst() {
+	if ! has_version ${CATEGORY}/${PN} ; then
+		elog "For correct logging add the following to /etc/syslog.conf:"
+		elog "    local6.*         /var/log/imapd.log"
+		elog "    auth.debug       /var/log/auth.log"
+		echo
+
+		elog "You have to add user cyrus to the sasldb2. Do this with:"
+		elog "    saslpasswd2 cyrus"
+	fi
+}
+
+pkg_postinst() {
+	# do not install server.{key,pem) if they exist
+	if use ssl ; then
+		if [[ ! -f "${ROOT}"/etc/ssl/cyrus/server.key ]]; then
+			install_cert /etc/ssl/cyrus/server
+			chown cyrus:mail "${ROOT}"/etc/ssl/cyrus/server.{key,pem}
+		fi
+	fi
+
+	if ver_replacing -lt $(ver_cut 1-2) ; then
+		elog "Please see https://www.cyrusimap.org/$(ver_cut 1-2)/imap/download/upgrade.html"
+		elog "for upgrade instructions."
+		if use backup ; then
```

Comment:
If a user installs this for the first time AND has enabled USE="backup" they won't see this depreciation message